### DISCUSSION TITLE
Dinero V2 release

### DISCUSSION THREAD

**Discussion by codams** [dinerojs/dinero.js#618] [General]
Hi there, 

_If this question was already answered, link me to it please ;)_ 

I'm interested in Dinero.js for comp projects and made a little POC with it, it fits very well our needs.
The API is clean and very well documented it is very pleasant to use !

I wanted to know if you have a **release date in mind for v2** ?

If help is needed for project maintaining I could be interested as well 👋

Have a great sunny day ☀️ !

**Comment by gerwinbrunner** [comment]
Would also be interested in the answer to this question

**Comment by johnhooks** [comment]
The majority of the api is solid and well tested, but there are a few important changes included in PR #309. It restructures the formatting of Dinero objects, adds support for non-decimal currencies and rounding options in `transformScale`(I am excited about this one).

I am not the maintainer, but I have been helping @sarahdayan move this forward. PR #309 is getting pretty close to completion. Once merged there aren't many obstacles to releasing v2.

**Reply by codams** [reply]
Noice

**Comment by sarahdayan** [comment]
We're on the right path. As @johnhooks said, #309 contains the biggest changes, after that it should be relatively smooth.

There are topics i'd like to explore before going beta because they might be breaking. I started tracking them in [this project](https://github.com/orgs/dinerojs/projects/3).

The board isn't exhaustive for now, but anything else should be relatively small.

Note that we'll go beta before going stable. This remains a strong commitment regarding the API, but gives us time to focus on helping current v1 users migrate to v2, and collect feedback regarding issues.

**Comment by johnhooks** [comment]
@sarahdayan something to consider is to leverage the `api-extractor` tags `@public`, `@beta` and `@alpha`.

There could be multiple releases. `api-extractor` can create multiple type declarations for the different release levels. It would take some more configuring of `api-extractor` but it would allow a release of everything you consider stable, and keep anything you felt could be breaking behind such as `@2.0.1`, `@2.0.1-beta.3`,  `@2.0.1-alpha.4`

It's just a thought. All the code would be the same and if someone dug around they could possibly use alpha functions on accident, but TypeScript would throw errors.

**Comment by johnhooks** [comment]
My opinions on the project structure are:

- Most everything can be moved to `dinero.js`. Having the api functions stretched between core and the main package creates a lot of repetitive code (especially if adding explicit return types, see PR #682 )
- All types and anything that could be useful to additional packages should remain in `@dinero.js/core`. If it's only types it could be listed as a dev dependancy.
- `calculator-number` should be moved to `dinero.js`, its is probably going to be everyones default, its always going to be the fastest, will work in every JS environment, and its very small.
- `calculator-bigint` could be moved to `dinero.js` again its very small and doesn't have much use outside of `dinero.js`. (Is compatiblity an issue? It wouldn't be for me, but might be for others.)
- Currencies should be separate packages, probably one for every desired currency type.

**Reply by sarahdayan** [reply]
I turned the topic from the project into an issue, can we take the conversation there so we keep track of it in the right place?

**Comment by osdiab** [comment]
@sarahdayan @johnhooks is this still planned to happen? looks like the PR mentioned above got merged but otherwise progress unclear. Thanks!

**Comment by sarahdayan** [comment]
Hey everyone, thanks for your patience and for checking in.

A lot has happened since the last update here. PR #309 was merged, and I've since consolidated all packages into a single `dinero.js` entry point (one of the bigger structural changes needed before a stable release). The docs site has also been revamped with a new design and updated content.

The remaining work is mostly documentation, a few API refinements, and final polish. The alpha is already solid for production use.

@codams, appreciate the offer to help! The best way to contribute right now is to try the alpha, report any issues, and share feedback.

I'm going to close this discussion and point you to #764, which is the canonical thread for v2 status updates. Feel free to follow along there!
